### ISSUE TITLE
error when using SKIP

### ISSUE THREAD

**Issue by rajgaire** [balderdashy/sails-mysql#237]
When limit and skip options are set, the query appends LIMIT x OFFSET y. When executing, mysql is returning empty set. This causes line 
    adapter.js (line 883)  cb(null, result[0].count); 
to fail as result = []

**Comment by devinivy** [comment]
Confirmed!  Let's get a failing test in the adapter tests.  And see if this is also a problem with sails-postgresql.  This query [here](https://github.com/balderdashy/sails-mysql/blob/master/lib/sql.js#L67-L70) needs to change from,
```sql
SELECT count(*) as count FROM table LIMIT 10 OFFSET 10;
```
to
```sql
SELECT count(*) as count FROM (SELECT * FROM table LIMIT 10 OFFSET 10) AS table;
```

**Comment by rajgaire** [comment]
@devinivy your suggested query will produce expected results. For example, if your table has 100 rows and you query
```SQL
    SELECT count(*) FROM table LIMIT 10;
```
you get 100. But if you query 
```SQL
    SELECT count(*) as count FROM (SELECT * FROM table LIMIT 10) AS table;
```
you will get 10.

Changing count query to your suggested query will break applications depending on it if they are relying on the mysql count behaviour.

OFFSET seems to be a problem of mysql. I would expect mysql ignores OFFSET in the same way as it does for LIMIT. 

From sails-mysql perspective, I would rather ignore 'skip' when querying count by adding 
```javascript
    delete options.skip;
```
in the __ COUNT __ function.

**Comment by devinivy** [comment]
I would suggest that this is in fact a bug that needs fixing.  When paging, how else would I count how many records might exist on a certain page?  Waterline's polyfill for `count` when not implemented by the adapter is to do a `find` then count the number of records returned.  That is more consistent with the fix I proposed than to ignore `skip`.

**Comment by particlebanana** [comment]
Fixed in waterline sequel `0.5.0`.
